### PR TITLE
bugfix: fixed the inability to get some remote configurations

### PR DESCRIPTION
**Pull request by a364176773**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
该bug会导致nacos等配置中心配置了store.mode=db,但是本地的openapi的实现类却用的file,除非在本地properties再配置上store.mode=db,否则不生效
而如果要通过nacos,需要先实例化fileconfiguration,而fileconfiguration会被springbootcglib代理,原先的实现用的是applicationcontext在bean加载后去findbean,而GenericApplicationListener的加载是非常早的,所以导致了configuration无法使用,特此需要改造使用ConfigurableEnvironment去读取配置,并转换

### CODE REVIEW

**Comment by a364176773** [comment]
@slievrly @wangliang181230 @xingfudeshi PTAL

File: common/src/main/java/io/seata/common/util/StringUtils.java [hunk 1]
```diff
@@ -319,4 +324,29 @@ public static boolean isEmpty(final CharSequence cs) {
...
+
+    /**
+     * hump to Line or line to hump
+     * 
+     * @param str str
+     * @return string string
+     */
+    public static String hump2Line(String str) {
+        Matcher matcher = HUMP_PATTERN.matcher(str);
+        StringBuffer sb = new StringBuffer();
```
Comment (xingfudeshi) [inline common/src/main/java/io/seata/common/util/StringUtils.java:336]: StringBuilder

**Review by xingfudeshi — APPROVED** [review]
LGTM

File: common/src/main/java/io/seata/common/util/StringUtils.java (hunk 1 above)
Comment (a364176773) [reply common/src/main/java/io/seata/common/util/StringUtils.java:336]: Matcher appendTail需要用StringBuffer

**Review by wangliang181230 — APPROVED** [review]
LGTM

File: common/src/main/java/io/seata/common/util/StringUtils.java [hunk 2]
```diff
@@ -36,6 +38,9 @@
  */
 public class StringUtils {
 
+    private static final Pattern HUMP_PATTERN = Pattern.compile("[A-Z]");
```
Comment (wangliang181230) [inline common/src/main/java/io/seata/common/util/StringUtils.java:41]: `hump` -> `camel` 吧？好像其他地方看到的都是 `camel` 这个单词。

Comment (a364176773) [reply common/src/main/java/io/seata/common/util/StringUtils.java:41]: ok

File: common/src/main/java/io/seata/common/util/StringUtils.java [hunk 3]
```diff
@@ -319,4 +324,29 @@ public static boolean isEmpty(final CharSequence cs) {
...
         return !isEmpty(cs);
     }
+
+    /**
+     * hump to Line or line to hump
+     * 
+     * @param str str
+     * @return string string
+     */
+    public static String hump2Line(String str) {
```
[the remark this answers is not in the document]
Comment (a364176773) [reply common/src/main/java/io/seata/common/util/StringUtils.java:334]: ok

File: common/src/main/java/io/seata/common/util/StringUtils.java (hunk 2 above)
Comment (a364176773) [reply common/src/main/java/io/seata/common/util/StringUtils.java:41]: done

File: common/src/main/java/io/seata/common/util/StringUtils.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (a364176773) [reply common/src/main/java/io/seata/common/util/StringUtils.java:334]: done

File: seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java [hunk 4]
```diff
@@ -85,25 +90,29 @@ private Object get(String dataId, Object defaultValue) throws IllegalAccessExcep
...
         Class<?> propertyClass = PROPERTY_BEAN_MAP.get(propertyPrefix);
         Object valueObject = null;
         if (propertyClass != null) {
             try {
-                Object propertyBean = applicationContext.getBean(propertyClass);
-                valueObject = getFieldValue(propertyBean, propertySuffix, dataId);
+                valueObject = getFieldValue(
+                    Objects.requireNonNull(PROPERTY_BEAN_INSTANCE_MAP.computeIfAbsent(propertyPrefix, k -> {
+                        try {
+                            return propertyClass.newInstance();
```
Comment (slievrly) [inline seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java:103]: The lack of bean injection is a problem for the framework. I don't think we should use the default instance.

File: common/src/main/java/io/seata/common/util/StringUtils.java [hunk 5]
```diff
@@ -319,4 +324,29 @@ public static boolean isEmpty(final CharSequence cs) {
...
+                matcher.appendReplacement(sb, "-" + matcher.group(0).toLowerCase());
+            }
+        } else {
+            matcher = LINE_PATTERN.matcher(str);
+            while (matcher.find()) {
+                matcher.appendReplacement(sb, matcher.group(1).toUpperCase());
+            }
+        }
+        matcher.appendTail(sb);
+        return sb.toString();
```
Comment (slievrly) [inline common/src/main/java/io/seata/common/util/StringUtils.java:349]: if all words are low case, return empty?

File: seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java [hunk 6]
```diff
@@ -118,19 +127,30 @@ private Object get(String dataId) throws IllegalAccessException, InstantiationEx
...
         if (fieldOptional.isPresent()) {
             Field field = fieldOptional.get();
+            Object value;
             field.setAccessible(true);
-            value = field.get(object);
-            if (value instanceof Map) {
-                String key = StringUtils.substringAfterLast(dataId, String.valueOf(DOT));
-                value = ((Map) value).get(key);
+            Object defaultValue = field.get(object);
+            if (defaultValue instanceof Map) {
```
Comment (slievrly) [inline seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java:139]: whether or not use field.getType()

Comment (a364176773) [reply seata-spring-autoconfigure/seata-spring-autoconfigure-core/src/main/java/io/seata/spring/boot/autoconfigure/provider/SpringBootConfigurationProvider.java]: done

File: common/src/main/java/io/seata/common/util/StringUtils.java (hunk 5 above)
Comment (a364176773) [reply common/src/main/java/io/seata/common/util/StringUtils.java:349]: matcher.appendTail(sb); 处理了，测试用例也测了这种情况

**Pull request merged**